### PR TITLE
Fix for failing kafka-0.8 test cases…

### PR DESCRIPTION
**Pull request by gsps1**
We skip exposing scala classes from system class loader to the program class loader for non-spark programs. 
Since the kafka flowlet program is non-spark, the program class loader doesn't have access to scala classes and is not able to load scala.Product class. 

we now get the scala jar and add it as additional jar during deploying program jar.

### CODE REVIEW

File: cdap-kafka-pack/cdap-kafka-flow/cdap-kafka-flow-core/src/test/java/co/cask/cdap/kafka/flow/KafkaConsumerFlowletTestBase.java [hunk 1]
```diff
@@ -116,10 +118,18 @@ public void cleanUpMetrics() throws Exception {
     return args;
   }
 
+  private ApplicationManager deployApplicationWithSparkJar() {
```
Comment (chtyim) [inline cdap-kafka-pack/cdap-kafka-flow/cdap-kafka-flow-core/src/test/java/co/cask/cdap/kafka/flow/KafkaConsumerFlowletTestBase.java]: Not Spark, Scala.

Comment (gsps1) [inline cdap-kafka-pack/cdap-kafka-flow/cdap-kafka-flow-core/src/test/java/co/cask/cdap/kafka/flow/KafkaConsumerFlowletTestBase.java]: fixed

**Comment by gokulavasan** [comment]
Hmm I see the change is made only in the test base. Does that mean the users of the kafka pack need to do something more than just depending on the cdap-kafka-compat-0.8?

**Comment by chtyim** [comment]
@gokulavasan It only affects unit-test, due to the way CDAP trace dependencies and build a temporary jar for unit-test deployment. In actual run time, the scala-core.jar is in the app-jar, hence this shouldn't be an issue.

**Comment by chtyim** [comment]
LGTM

**Comment by gokulavasan** [comment]
@gsps1 This means that the unit test in kafka ingest guide might have to be updated too?

**Comment by gsps1** [comment]
@gokulavasan good call, yea that will fail too. i will send a fix for that as well.

**Pull request merged**